File: pinyougou-manager-web/src/main/java/com/lsf/pinyougou/manager/controller/BrandController.java
package com.lsf.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.lsf.pinyougou.pojo.TbBrand;
import com.lsf.pinyougou.sellergoods.service.BrandService;
import org.springframework.core.env.Environment;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import vo.PageResult;
import vo.Result;

import java.util.List;
import java.util.Map;


/**
 * 运营商管理后台-品牌列表控制器
 *
 * @RestController 是 Responsebody + Controller 的组合
 */
@RestController
@RequestMapping("/brand")
public class BrandController {


    /**
     * 查询所有品牌
     *
     * 此处返回的集合会被 springMVC 管理的 fastJSON 转换器转换为 json 数据传给 Web 消费方
     */
    @RequestMapping("/findAll.do")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }


    /**
     * 品牌分页
     *
     * @param page  当前页码
     * @param size  每页记录数
     */
    @RequestMapping("/findPage.do")
    public PageResult findPage(int page, int size) {
        return brandService.findPage(page, size);
    }


    /**
     * 多条件品牌分页查询
     */
    @RequestMapping("/search.do")
    public PageResult findPageLimit(@RequestBody TbBrand brand, int page, int size) {
        return brandService.findPageLimit(brand, page, size);
    }


    /**
     * 添加品牌
     *
     * @param brand 前端传递的封装对象,需要使用 @RequestBody 注解标记
     * @return  Result 是给前端的响应体
     */
    @RequestMapping("/add.do")
    public Result add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            return new Result(true, "添加成功");
        } catch (Exception e) {
            return new Result(false, "添加失败");
        }
    }


    /**
     * 根据品牌 ID 查询品牌信息
     */
    @RequestMapping("/findOne.do")
    public TbBrand findOne(long id) {
        return brandService.findOne(id);
    }


    /**
     * 修改品牌信息
     */
    @RequestMapping("/update.do")
    public Result update(@RequestBody TbBrand brand) {
        try {
            brandService.update(brand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            return new Result(false, "修改失败");
        }
    }


    /**
     * 批量删除品牌
     */
    @RequestMapping("/batchDelete.do")
    public Result batchDelete(Long[] ids) {
        try {
            brandService.batchDelete(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            return new Result(false, "删除失败");
        }
    }


    /**
     * 查询所有品牌的列表
     */
    @RequestMapping("/selectBrandList.do")
    public List<Map> selectBrandList() {
        try {
            return brandService.selectBrandList();
        } catch (Exception e) {
            return null;
        }
    }


    @Reference
    private BrandService brandService;
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
// 品牌列表管理控制器 brandController,该控制器依赖品牌列表服务模块
pyg.controller('brandController', function ($scope, $controller, brandService) {

    // 继承父类控制器 baseController
    $controller('baseController', {$scope:$scope});


    // 在控制器中添加查询所有品牌列表的方法
    $scope.findAll = function () {
        // 在控制器中调用品牌列表服务模块的功能
        brandService.findAll().success(
            function (response) {
                // response 就是后台返回的 json 格式存储的品牌列表数据
                $scope.brandList = response;
            }
        );
    };


    // 添加品牌分页查询的方法
    $scope.findPage = function (page, size) {
        brandService.findPage(page, size).success(
            function (response) {
                // 修改当前页数据
                $scope.brandList = response.rows;
                // 修改总记录数
                $scope.paginationConf.totalItems = response.total;
            }
        );
    };

    // 定义搜索对象
    $scope.searchEntity = {};

    // 多条件品牌分页查询
    $scope.search = function (page, size) {
        brandService.findPageLimit(page, size, $scope.searchEntity).success(
            function (response) {
                // 修改当前页数据
                $scope.brandList = response.rows;
                // 修改页码
                $scope.paginationConf.currentPage = page;
                // 修改总记录数
                $scope.paginationConf.totalItems = response.total;
            }
        );
    };

    // 保存品牌(可能是新增品牌，可能是修改品牌)
    $scope.save = function () {
        var object = null;
        if ($scope.brandEntity.id != null) {
            // id 不为空，说明是修改品牌
            object = brandService.update($scope.brandEntity);
        } else {
            object = brandService.add($scope.brandEntity);
        }

        object.success(
            function (response) {
                if (response.success) {
                    // 保存成功
                    // 刷新当前页
                    $scope.reloadList();
                } else {
                    // 保存失败, 提示错误信息
                    alert(response.message);
                }
            }
        );
    };

    // 根据品牌 ID 查询品牌信息
    $scope.findOne = function (id) {
        brandService.findOne(id).success(
            function (response) {
                // 将查询到的数据显示在修改框中
                $scope.brandEntity = response;
            }
        );
    };


    // 监听删除按钮
    $scope.batchDelete = function () {
        if ($scope.selectedList.length === 0) {
            alert("请选择要删除的内容");
        } else {
            brandService.batchDelete($scope.selectedList).success(
                function (response) {
                    if (response.success) {
                        // 删除成功
                        // 刷新当前页
                        $scope.reloadList();
                    } else {
                        // 删除失败
                        alert(response.message);
                    }
                });

            // 删除成功后，要清空集合
            $scope.selectedList = [];
        }
    };
});